File: php/leet_code/1614.php
<?php

    class Solution
    {

        /**
         * @param String $s
         * @return Integer
         */
        function maxDepth($s)
        {
            $max = 0;
            $depth = 0;
            for ($i = 0, $iMax = strlen($s); $i < $iMax; $i++) {
                if ($s[$i] === '(') {
                    $depth++;
                    $max = max($max, $depth);
                } elseif ($s[$i] === ')') {
                    $depth--;
                }
            }
            return $max;
        }
    }

File: swift/leet_code/1614.swift
class Solution {
    func maxDepth(_ s: String) -> Int {
        var maxDepth = 0
        var depth = 0
        for char in s {
            if char == "(" {
                depth += 1
            } else if char == ")" {
                depth -= 1
            }
            maxDepth = max(maxDepth, depth)
        }
        return maxDepth
    }
}
